### ISSUE TITLE
Query string parameter validation of Arrays with `style: form` and `explode: false`

### ISSUE THREAD

**Issue by digilist** [lezhnev74/openapi-psr7-validator#38]
Consider this swagger parameter definition (in YAML):
```yaml
    -
        name: filter
        in: query
        description: Filter something
        explode: false
        style: form
        schema:
            type: array
            items:
                type: string
```

According to the [Open Api Specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#style-examples) this will result in a query parameter  `?filter=a,b,c` (which is also generated by Swagger UI)

However, the validation does not consider a URL of that format as an array, but as a string:

```
OpenAPI spec does not match the query argument 'with' of the request [/test,get]: Value expected to be 'array', 'string' given.
```

**Comment by lezhnev74** [comment]
Yeah, true. At this point, parameters serialization and deserialization are not supported at all in this package.

**Comment by NullNinja1** [comment]
> Yeah, true. At this point, parameters serialization and deserialization are not supported at all in this package.

I also have ran into this issue today.

Would this not be something that could be possibly fixed with the changes that were added here just recently?  -> https://github.com/lezhnev74/openapi-psr7-validator/issues/36

**Comment by lezhnev74** [comment]
Well, serialization is not so straightforward to me, as there are many supported serialization patterns: https://swagger.io/docs/specification/serialization/

#36 incorporates much simpler things like typecasting (it uses just `type` keyword and basically validates the query argument based on that).

**Comment by judgej** [comment]
In case it is of any use for ideas, we are generating clients from OpenAPI descriptions using the OpenAPI Code Generation project. Are using heavily customised templates (and still WIP) but this function converts an array of strings passed into an operation method, into a serialised parameter string suitable for a query parameters:

https://github.com/consilience/xero-api-sdk/blob/master/apis/accounting/src/ObjectSerializer.php#L209

This is one big we haven't touched yet, so probably needs some work, but I'm guessing deserialising a parameter will be the reverse of this. The serialisation looks fairly simple on the face of it, at least for query parameters (path parameters though, yikes).

Would takling simple serialisation on just query parameters, effectively casting them to arrays using different joining patterns, be a relatively simple first step? I *suspect* that will be the most common use-case anyway.

**Comment by lezhnev74** [comment]
closed in favor of #47
